.github/workflows/conformance-kubespray.yaml

```diff
@@ -0,0 +1,205 @@
+
+          [kube_control_plane]
+          localhost
+
+          [etcd]
+          localhost
+
+          [kube_node]
+          localhost
+
+          [k8s_cluster:children]
+          kube_control_plane
+          kube_node
+          EOF
+
+          sed -e 's/^kube_network_plugin:.*$/kube_network_plugin: cni/' \
+              -e 's/^kube_owner:.*$/kube_owner: root/' \
+              inventory/mycluster/group_vars/k8s_cluster/k8s-cluster.yml > k8s-cluster.tmp
+          mv k8s-cluster.tmp inventory/mycluster/group_vars/k8s_cluster/k8s-cluster.yml
+
+      - name: Deploy Kubernetes cluster
+        run: |
+          source kubespray-venv/bin/activate
+          cd kubespray/
+          ansible-playbook -i inventory/mycluster/inventory.ini cluster.yml -b -v
+
+          mkdir -p $HOME/.kube
+          sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+          sudo chown "$(id -u)":"$(id -g)" $HOME/.kube/config
+
+          kubectl wait --for=condition=Ready pods --all -n kube-system --timeout=300s
+
+      - name: Set up job variables
+        id: vars
+        run: |
+          CILIUM_INSTALL_DEFAULTS="${{ steps.default_vars.outputs.cilium_install_defaults }} \
+            --helm-set=cluster.name=${{ env.clusterName }}"
+          CONNECTIVITY_TEST_DEFAULTS="--test-concurrency=5 \
+            --test='!/pod-to-world'"
+          echo cilium_install_defaults=${CILIUM_INSTALL_DEFAULTS} >> $GITHUB_OUTPUT
+          echo connectivity_test_defaults=${CONNECTIVITY_TEST_DEFAULTS} >> $GITHUB_OUTPUT
+          echo sha=${{ steps.default_vars.outputs.sha }} >> $GITHUB_OUTPUT
+
+      - name: Install Cilium CLI
+        uses: cilium/cilium-cli@e7f03e4ec38a7008f4f5b9855ca9df721a9db185 # v0.18.3
+        with:
+          skip-build: ${{ env.CILIUM_CLI_SKIP_BUILD }}
+          image-repo: ${{ env.CILIUM_CLI_IMAGE_REPO }}
+          image-tag: ${{ steps.vars.outputs.sha }}
+          repository: ${{ env.CILIUM_CLI_RELEASE_REPO }}
+          release-version: ${{ env.CILIUM_CLI_VERSION }}
+
+      # Warning: since this is a privileged workflow, subsequent workflow job
+      # steps must take care not to execute untrusted code.
+      - name: Checkout pull request branch (NOT TRUSTED)
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ steps.vars.outputs.sha }}
+          persist-credentials: false
+          path: untrusted
+          sparse-checkout: |
+            install/kubernetes/cilium
+
+      - name: Install Cilium
+        id: install-cilium
+        run: |
+          cilium install ${{ steps.vars.outputs.cilium_install_defaults }}
+
+      - name: Wait for Cilium to be ready
+        run: |
+          cilium status --wait --interactive=false --wait-duration=2m
+          kubectl get pods --all-namespaces -o wide
+
+      - name: Run connectivity test
+        run: |
+          cilium connectivity test ${{ steps.vars.outputs.connectivity_test_defaults }}
```

Comment:
Most workflows in Cilium have additional steps after this. For instance steps that will gather test artifacts like a sysdump for further inspection of test results. After https://github.com/cilium/cilium/pull/39787 this should get a lot easier actually.

Could you share which workflow you used as a basis for writing this workflow? That could help reviewers to understand the differences here and whether we might still want some other of the general boilerplate.

Reply:
Thanks! I referenced `.github/workflows/conformance-kind-proxy-embedded.yaml`.

I saw the common steps, but implementations vary across other conformance workflows (AWS, AKS, EKS), so I wasn't sure which were essential and omitted them for now. I am happy to add any you think are critical.

Reply:
Hmm, those last steps should be pretty consistent in general, other than that some of the cloud conformance workflows may either (a) have a cloud-specific cleanup step, or (b) may have slightly different matrix configurations.

Overall those end steps are there to try to provide triagers with consistent information such as a dump of all of the state in the cluster to debug further, as well as summaries of features used and the results themselves. The fewer differences we have between the different workflows, the better IMO just for a more consistent experience when interacting with the workflows.